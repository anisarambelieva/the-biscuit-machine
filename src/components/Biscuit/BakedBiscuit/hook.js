const useBakedBiscuit = (onBaked) => {
  let animationsCount = 0;

  const handleAnimationEnd = () => {
    animationsCount++;

    if (animationsCount % 3 === 0) {
      onBaked(false);
    }
  };

  return handleAnimationEnd;
};

export default useBakedBiscuit;
