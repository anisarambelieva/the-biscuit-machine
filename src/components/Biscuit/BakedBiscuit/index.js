import "./style.css";
import useBakedBiscuit from "./hook.js";

const BakedBiscuit = ({ state, onBaked }) => {
  const handleAnimationEnd = useBakedBiscuit(onBaked);

  return (
    <div
      onAnimationEnd={handleAnimationEnd}
      className="animatedBakedBiscuit"
      style={{
        WebkitAnimationPlayState: state,
      }}
    >
      Baked
    </div>
  );
};

export default BakedBiscuit;
