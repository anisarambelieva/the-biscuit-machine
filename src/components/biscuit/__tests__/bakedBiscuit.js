import { render } from "@testing-library/react";

import BakedBiscuit from "../bakedBiscuit.js";

describe("<BakedBiscuit />", () => {
  it("should match snapshot", () => {
    const { container } = render(<BakedBiscuit />);

    expect(container).toMatchSnapshot();
  });
});
