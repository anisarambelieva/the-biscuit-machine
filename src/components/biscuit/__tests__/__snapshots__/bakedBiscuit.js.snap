// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BakedBiscuit /> should match snapshot 1`] = `
<div>
  <div
    class="bakedBiscuit"
    style="margin: 65px 0px 0px 0px; position: absolute; padding-left: 25px; z-index: -1;"
  >
    Baked
  </div>
</div>
`;
